signer/src/wsts_state_machine.rs

```diff
@@ -45,7 +45,19 @@ pub enum StateMachineId {
     /// Identifier for a Bitcoin signing state machines
     BitcoinSign(SigHash),
     /// Identifier for a rotate key verification signing round
-    RotateKey(PublicKeyXOnly, model::BitcoinBlockHash),
+    RotateKey(PublicKeyXOnly),
```

Comment:
Are we sure about removing the bitcoin block hash (in this moment)? It may have effects we don't want to handle now, eg having old state machines around in some state that may be reused in a new tenure (validating the same aggregate key), potentially stalling the process.

Reply:
Well it's supposed to work in tandem with the internal state machine timeout, as after that timeout the state machine will be invalid. I changed the timeout to the config value for the signing timeout that the coordinator uses for all signing (including this one), which by default is 10s. The tx-signer also resets the state machine on retrieval if it exists and is in an end-state/timed-out now, and also removes its sibling signer-state-machine.

Idunno, I don't think there's anything "bad" that can happen (short of maybe one tenure erroring), but I could have missed an edge case. I'm still missing tests for this logic, though.

Reply:
To me it seems like a new complexity we need to be sure works as expected with the rest of changes (and the already funky set of timeouts and interactions we have), so unless it's something we need right now I would prefer to keep the current _stick to the coordinator tenure_ behaviour and change it later on, instead of a mix situation of having some of the changes for cross-tenure things but still everything else tied to the coordinator tenure.

Reply:
I'm fine with adding block hash back, I'm genuinely not that concerned about these signing transactions crossing bitcoin block boundaries as we've never really had a problem with pure signing (on the individual-round level, anyway); and if it happens we have the other safeguards & fallbacks.

Reply:
If the new coordinator wouldn't try again I could agree, but it could happen that flashblocks + new coordinator doing new coordinator things would put us in a strange situation (maybe with some state machine still dirty and not in a terminal state yet) -- which could be fine, but I'm not sure and that's kinda my point: this whole DKG verification is already adding complexity, I would really avoid adding now anything extra that is not needed and we are not sure is not harmful.

Reply:
> I'm fine with adding block hash back

☝️ :) I was just adding sub-thoughts -- I'll add it back

Reply:
Done in [2320ef9](https://github.com/stacks-network/sbtc/pull/1324/commits/2320ef9061541789bd5b4e23734b402444426807), though it became a `BlockHashRef`, same as the `StateMachineId::Dkg` variant uses now.